nodecg-io-midi-input/extension/index.ts

```diff
@@ -16,12 +16,34 @@ module.exports = (nodecg: NodeCG) => {
 
 class MidiService extends ServiceBundle<MidiInputServiceConfig, MidiInputServiceClient> {
     async validateConfig(config: MidiInputServiceConfig): Promise<Result<void>> {
-        new easymidi.Input(config.device).close();
+        const devices: Array<string> = new Array<string>();
+
+        easymidi.getInputs().forEach((device) => {
+            if (device.includes(config.device)) {
+                devices.push(device);
+            }
+        });
+
+        if (devices.length == 0) {
+            return error("no device matched the configured pattern.");
+        }
+        if (devices.length > 1) {
+            return error("The configured pattern is ambiguous.");
+        }
         return emptySuccess();
     }
 
     async createClient(config: MidiInputServiceConfig): Promise<Result<MidiInputServiceClient>> {
-        this.nodecg.log.info(`Connecting to MIDI input device ${config.device}.`);
+        this.nodecg.log.info(`Checking device name "${config.device}"`);
+        const devices: Array<string> = new Array<string>();
+        let deviceName: string | null = null;
+        easymidi.getInputs().forEach((device) => {
+            if (device.includes(config.device) && deviceName == null) {
+                deviceName = device;
+            }
+        });
+
+        this.nodecg.log.info(`Connecting to MIDI input device ${deviceName}.`);
         const client = new easymidi.Input(config.device);
         this.nodecg.log.info(`Successfully connected to MIDI input device ${config.device}.`);
```

Comment:
We should also use the found device here and not the maybe incomplete string that the user has entered.
```suggestion
        const client = new easymidi.Input(deviceName);
        this.nodecg.log.info(`Successfully connected to MIDI input device ${deviceName}.`);
```
You also need to check `deviceName` for null. The framework doesn't call validate when loading the config from disk because it assumes that a config which was once valid will be always valid. This was done to improve startup performance but I'm thinking of changing it back because it creates this hassle for service devs.

Reply:
I'll fix that for now.

Reply:
added a little check to enshure this is not going to break.